tests/pipeline/test_factor.py

```diff
@@ -553,3 +649,191 @@ class DateFactor(Factor):
         ).format(normalizer=method_name)
 
         self.assertEqual(errmsg, expected)
+
+    @parameter_space(seed=[1, 2, 3])
+    def test_quantiles_unmasked(self, seed):
+        permute = partial(permute_rows, seed)
+
+        shape = (6, 6)
+
+        # Shuffle the input rows to verify that we don't depend on the order.
+        # Div by 2 to ensure that we don't depend on inputs being integral.
+        factor_data = permute(arange(36, dtype=float).reshape(shape)) / 2.0
+
+        f = self.f
+        terms = {
+            '2': f.quantiles(bins=2),
+            '3': f.quantiles(bins=3),
+            '6': f.quantiles(bins=6),
+        }
+
+        # Apply the same shuffle we applied to the input rows to our
+        # expectations. Doing it this way makes it obvious that our
+        # expectation corresponds to our input, while still testing against
+        # a range of input orderings.
+        permuted_array = compose(permute, partial(array, dtype=int))
+        expected = {
+            # The values in the input are all increasing, so the first half of
+            # each row should be in the bottom bucket, and the second half
+            # should be in the top bucket.
+            '2': permuted_array([[0, 0, 0, 1, 1, 1],
+                                 [0, 0, 0, 1, 1, 1],
+                                 [0, 0, 0, 1, 1, 1],
+                                 [0, 0, 0, 1, 1, 1],
+                                 [0, 0, 0, 1, 1, 1],
+                                 [0, 0, 0, 1, 1, 1]]),
+            # Similar for three buckets.
+            '3': permuted_array([[0, 0, 1, 1, 2, 2],
+                                 [0, 0, 1, 1, 2, 2],
+                                 [0, 0, 1, 1, 2, 2],
+                                 [0, 0, 1, 1, 2, 2],
+                                 [0, 0, 1, 1, 2, 2],
+                                 [0, 0, 1, 1, 2, 2]]),
+            # In the limiting case, we just have every column different.
+            '6': permuted_array([[0, 1, 2, 3, 4, 5],
+                                 [0, 1, 2, 3, 4, 5],
+                                 [0, 1, 2, 3, 4, 5],
+                                 [0, 1, 2, 3, 4, 5],
+                                 [0, 1, 2, 3, 4, 5],
+                                 [0, 1, 2, 3, 4, 5]]),
+        }
```

Comment:
Would it be worth testing the case where number of bins does not evenly divide the number of columns? i.e. if your factor had 4 columns and you wanted 3 bins, should that give this?
```python
[[0, 0, 1, 2],
 [0, 0, 1, 2],
...
```

Or, if you ask for more bins than there are columns, should that raise a certain error?